File: Php/app/Urls/CashgameChartUrlModel.php
<?php
namespace app\Urls{

	use app\RouteFormats;
	use app\Urls\BaseClasses\HomegameYearUrlModel;
	use entities\Homegame;

	class CashgameChartUrlModel extends HomegameYearUrlModel{

		public function __construct(Homegame $homegame, $year){
			parent::__construct(RouteFormats::cashgameChart, RouteFormats::cashgameChartWithYear, $homegame, $year);
		}

	}

}

File: Web/Models/UrlModels/CashgameChartUrlModel.cs
using Web.Routing;

namespace Web.Models.UrlModels
{
    public class CashgameChartUrlModel : HomegameWithOptionalYearUrlModel
    {
        public CashgameChartUrlModel(string slug, int? year)
            : base(RouteFormats.CashgameChart, RouteFormats.CashgameChartWithYear, slug, year)
        {
        }
    }
}